qiskit_machine_learning/algorithms/classifiers/pegasos_qsvc.py

```diff
@@ -0,0 +1,276 @@
+        """
+        if not self._fit_status:
+            raise QiskitMachineLearningError("The PegasosQSVC has to be fit first")
+        if np.ndim(X) != 2:
+            raise ValueError("X has to be a 2D array")
+        if self._precomputed and self._n_samples != X.shape[1]:
+            raise ValueError(
+                "For a precomputed kernel, X should be in shape (m_samples, n_samples)"
+            )
+
+        t_0 = datetime.now()
+        y = np.zeros(X.shape[0])
+        for i in range(X.shape[0]):
+            value = self._compute_weighted_kernel_sum(i, X, training=False)
+
+            if value > 0:
+                y[i] = self._label_pos
+            else:
+                y[i] = self._label_neg
+
+        logger.debug("prediction completed after %s", str(datetime.now() - t_0)[:-7])
+
+        return y
+
+    def _compute_weighted_kernel_sum(self, index, X, training) -> float:
+        """Helper function to compute the weighted sum over support vectors used for both training
+        and prediction with the Pegasos algorithm.
+
+        Args:
+            index: fixed index distinguishing some datum
+            X: Features
+            training: flag indicating whether the loop is used within training or prediction
+
+        Returns:
+            float: weighted sum of kernel evaluations employed in the Pegasos algorithm
+        """
+        kernel: Dict[Tuple, np.ndarray] = {}
+        # this value corresponds to a sum of kernel values weighted by their labels and alphas
+        value = 0.0
+        # only loop over the non zero alphas (preliminary support vectors)
+        for j in self._alphas:
+            # perform the kernel evaluations on the fly, as intended
+            if not self._precomputed:
+                # for training
+                if training:
+                    x_j = X[j]
+                # for prediction
+                else:
+                    x_j = self._x_train[j]
+
+                # evaluate kernel function only for the fixed datum and the data with non zero alpha
+                kernel[(index, j)] = kernel.get(
+                    (index, j), self._quantum_kernel.evaluate(X[index], x_j)
+                )
+                kernel_value = kernel[(index, j)]
+
+            # consider a precomputed kernel
+            else:
+                kernel_value = X[index, j]
+
+            value += (
+                # alpha weights the contribution of the associated datum
+                self._alphas[j]
+                # the class membership labels have to be in {-1, +1}
+                * self._label_map[self._y_train[j]]
+                # the offset to the kernel function leads to an implicit bias term
+                * (kernel_value + self._kernel_offset)
+            )
+
+        return value
+
+    @property
+    def quantum_kernel(self) -> QuantumKernel:
+        """Returns quantum kernel"""
+        return self._quantum_kernel
+
+    @quantum_kernel.setter
```

Comment:
should training status (and other values) be reset when the kernel is replaced?

Reply:
See 258cabfa6f609159cc33d5ee94f2b899c45e7a91